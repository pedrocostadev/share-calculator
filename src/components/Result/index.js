import { connect } from 'react-redux';

import Result from './result';
import { operationsSelectors } from '../../selectors';

const mapStateToProps = state => ({
  text: operationsSelectors.selectResult(state),
});

export default connect(mapStateToProps)(Result);
