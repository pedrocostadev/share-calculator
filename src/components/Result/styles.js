import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  result: {
    fontSize: 28,
    color: '#FFFFFF',
    padding: 10,
    textAlign: 'right',
  },
});

export default styles;
