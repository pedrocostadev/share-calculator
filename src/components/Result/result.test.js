import React from 'react';
import Result from './result';
import { render, cleanup } from '@testing-library/react-native';

describe('<Result />', () => {
  afterEach(cleanup);

  test('should render', async () => {
    const text = '12.2';

    const { queryAllByText, container } = render(<Result text={text} />);
    const ocurrences = await queryAllByText(text);

    expect(ocurrences).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
