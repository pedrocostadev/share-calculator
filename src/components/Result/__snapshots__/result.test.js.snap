// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 28,
        "padding": 10,
        "textAlign": "right",
      }
    }
  >
    12.2
  </Text>
</View>
`;
